cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -536,6 +538,9 @@ public void onClose(KubernetesClientException e) {
                                 });
                             } else {
                                 log.info("{}: Getting Cruise Control rebalance user task status", reconciliation);
+                                // TODO: If this request results in an error from the server then the Rebalance state
+                                //       will switch to NotReady -> ProposalPending -> ProposalReady even though we are in
+                                //       an ongoing rebalance.
```

Comment:
Is this likely to happen? (Did you run into this issue while testing?)

Reply:
Yes, I miss configured the user tasks URL and got a 404 error during a check on the status of an ongoing rebalance. I fixed that, so that shouldn't be an issue. However, if there is a networking issue in k8s, or the CC API code is temp offline for some reason then we will get an error response and the Rebalance Resource well end up in a `ProposalReady` state during an ongoing rebalance. Which is obviously not what we want. We should not assume the network is infallible.

Reply:
I like the way you think Tom. Defensive programming is the way to go. I am ok with fixing this after the initial PR, I think users will have bigger problems with their clusters before this one if they have networking issues.